sdk/key_vault/src/key.rs

```diff
@@ -498,12 +332,48 @@ pub struct AesGcmDecryptParameters {
     pub additional_authenticated_data: Option<Vec<u8>>,
 }
 
+impl AesGcmParameters {
+    pub fn new(
+        algorithm: EncryptionAlgorithm,
+        iv: Vec<u8>,
+        authentication_tag: Vec<u8>,
```

Comment:
Because of these two - Microsoft Crypto Board has required we don't accept `iv` for input to encrypt, and you can only provide `authentication_tag` for decrypt, can you rename these to `*DecryptParameters`? The inputs for encrypt and decrypt are different and this will at least make it clear.

Reply:
done!